src/pages/index.js

```diff
@@ -67,250 +74,229 @@ const cardList = new Section({
-  //const card = new Card(cardData, '.card-template', handleImageClick)
-
-  const newCard = getCardView({
-    name,
-    link //since this is an object that's destructured  the order doesnt matter , if its an array the order matters
-  }); // im passing the data from here to the getCardView
-  /* const cardElement = getCardView({
-     name,
-     link
-   })
-     */
-  //call render card function and pass an object with name and link
-
-  //card.getElementView()
-  //const cardView = getCardView({
-  //  name, // name:name, since both vars and properties are called the same you can just put em once
-  //  link //link:link     you are setting the new variables with the attribute value assigned
-  // })
-  /*renderCard({name:title,link:link})*/
-  //card section
-  cardList.addItem(newCard)
-  //renderCard(newCard) // this invokes the rendering function with new variable and which will be appended into the gallery
-  addCardForm.close()
-  //closePopUp(cardAddModal) //close after the process
-
-  //e.target.reset(); //reset the card after inputting
-  //newPlaceBtn.disabled = true;
-  addFormValidator.disabledButton();
-  // search for button, disable it, add a class would've worked as well
+  editProfileForm.close();
+  //closePopUp(profileEditModal);
 }
 
-/*
-profileEditForm.addEventListener("submit", (e) => {
-  e.preventDefault(); // now i can add multple lines of code regarding the event handler by the use of arrow function.
-  handleProfileEditSubmit() // i could put e to pass it to the above function but i wouldn't need to now
-});
-*/
-profileEditBtn.addEventListener("click", handleEditButton);
+function handleCardAddSubmit(newCardData) {
 
- /*()=>{the body of toggle edit button could go here} */
-/*profileModalCloseBtn.addEventListener("click", () => {
-  closePopUp(profileEditModal); /* you could delete the arrow function and just use the name of closePopup as reference*/
-//});
+  const name = newCardData.title; //e.target.title.value // title is the name attribute
+  const link = newCardData.link; //e.target.link.value
 
 
-//cardAddForm.addEventListener('submit', handleCardAddSubmit)
 
-cardAddBtn.addEventListener('click', handleAddButton);
-/*
-cardAddCloseButton.addEventListener('click', () => {
-  //closePopUp(cardAddModal)
-})
-*/
+  api.postNewCard(name, link).then((res) => { //creating the card in the server ,res is the object with all the properties in the server
+    const newCard = getCardView(
+      /*{
+              name,
+              link,
+            }*/
+      res); //just on user interface
+    cardList.addItem(newCard);
+    addCardForm.close();
+  }).catch(error => console.error(error));
 
-//pictureModalCloseBtn.addEventListener('click', () => {
- // imagePopup.close()
-  //closePopUp(pictureModal)
-//}) //Close icons should be handled only once in the file body, otherwise you add listeners again and again to the same elements. This can cause a memory leak
 
+  addFormValidator.disabledButton();
```

Comment:
`  addFormValidator.disabledButton();
`
You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.